includes/class-wc-abstract-google-analytics-js.php

```diff
@@ -114,11 +114,17 @@ abstract protected function add_transaction_enhanced( $order );
 	 * @return string
 	 */
 	public static function get_product_identifier( $product ) {
-		if ( ! empty( $product->get_sku() ) ) {
-			return esc_js( $product->get_sku() );
-		} else {
-			return esc_js( '#' . ( $product->is_type( 'variation' ) ? $product->get_parent_id() : $product->get_id() ) );
+		$identifier = $product->get_id();
+
+		if ( 'product_sku' === self::get( 'ga_product_identifier' ) ) {
```

Comment:
Maybe add some constant/getter for `product_sku` literal?

Reply:
As the value won't change and how things are written elsewhere in the extension I've opted to leave as is for the sake of simplicity and readability.